patches/minecraft/net/minecraft/tags/Tag.java.patch

```diff
@@ -1,8 +1,13 @@
 --- a/net/minecraft/tags/Tag.java
 +++ b/net/minecraft/tags/Tag.java
-@@ -49,6 +_,7 @@
+@@ -47,8 +_,11 @@
+       return SetTag.m_13222_(p_13301_);
+    }
  
-    public static class Builder {
+-   public static class Builder {
++   public static class Builder implements net.minecraftforge.common.extensions.IForgeRawTagBuilder {
++      private final List<Tag.BuilderEntry> removeEntries = Lists.newArrayList(); /** Forge: Get the list of remove entries for datagen **/
++      public List<Tag.BuilderEntry> getRemoveEntries() { return this.removeEntries; } // also forge
        private final List<Tag.BuilderEntry> f_13302_ = Lists.newArrayList();
```

Comment:
Missed this in my last pass, this should return either a read only view, or like the vanilla getEntries, a stream. So that user's can't modify the value of the data objects without us knowing about it. Yes I know this means you'll need to move the implementation of one of the remove() functions to the patch. This is fine.